### ISSUE TITLE
Lccal Rework Objectives

### ISSUE THREAD

**Issue by johnridge** [rotsehub/rotseana#12]
This is a summary of the principal objectives of the lccal.py rework that @GrantPDonnelly and I currently working on. @rkehoe and @gdhungana, feel free to suggest additional changes or features.
1. Prevent potential variable stars from being used as reference stars
2. Filter reference star data prior to the computation of corrections
3. Add an argument to filter data with unconex without calibrating it
4. Remove the need to have lccal.py in the parent directory of the directory containing the match structures
5. Optional radius argument to enable or disable iterative expansion of search radius
6. Simplify, streamline, and compact underlying code
7. Improve code readability with meaningful object names and in-text comments
8. Increase execution speed by reworking the reference star search algorithm

**Comment by rkehoe** [comment]
@johnridge  This is a good list.  Can you indicate which Pull Request #10 or #11 this concerns?  I want us to avoid starting another PR and branch and resolve existing PRs as soon as possible.  Otherwise, the code development becomes confusing.  Please clarify what branch/PR these changes are meant for.

Since we need to converge on a viable version of lccal this week, I would encourage prioritization of these tasks.  Some can be done in a subsequent update after the next merge to master.  My suggested prioritization is:

Highest priority, for this week: 1 and 3
Next update (new branch and PR) after this week:  2 and 5
Low priority (after June 20th): 4, 6-8

Please clarify your plan for PRs on these tasks.  This usually involves a list of PRs and a reasonable timetable and time order.

**Comment by rkehoe** [comment]
I should have also done @GrantPDonnelly on my last comment.  Please confer with John about how to proceed and put a follow up clarifying comment.

**Comment by johnridge** [comment]
@rkehoe, to clarify the PR situation, this is not related to either PR #10 or #11. #11 is ready to be merged and #10 should be ready once #11 has been merged. I posted this issue so there would be an accessible record of our goals and objectives outside of our own personal notes and it was not meant to be associated with a specific PR.

**Comment by rkehoe** [comment]
Okay, @johnridge , @GrantPDonnelly, then I think this describes work past PRs #10 and #11.  In that case, please specify what subset of tasks in your list will go to an immediate PR after #10 and #11, and what will go later.  One of the uses of Issues is to prioritize and reorganize code development efforts.  I would not advise doing all 8 items in the next branch/PR.  So please suggest how you proceed in some detail.

Also, I think then we need to clear #10 and #11 today if possible.  Do I need to review them for that?

**Comment by johnridge** [comment]
@rkehoe, as I said #11 is ready to be merged. The change to the code is relatively small and I have confirmed that it functions as intended. I would have already merged it if I had write permissions. #10 will just merge integrated lccal/unconex into master alongside the small fix I made in #11. Extensive testing has not revealed any issues that should prevent it from being merged. That being said, #11 needs to be merged before #10 otherwise my fix will not get merged into master, which would cause problems.

**Comment by rkehoe** [comment]
Thanks @johnridge, this is clearer now.  I will merge #10 first, and then #11 

In general, we want to adhere to a coding model where we produce a branch off of master for a well-defined task, and then implement and test it.  Once it is ready, we merge into master before starting modifications on the same code modules in another branch.  So we make a branch, implement, test, merge, make another branch, implement, test, merge.  I understand how we got to the current situation, and that’s fine.  My comments here are meant constructively for the future updates.

**Comment by johnridge** [comment]
@rkehoe, just to clarify, #11 needs to be merged prior to #10. Your comment states the opposite.

**Comment by rkehoe** [comment]
@johnridge  What is the status of this Issue?

**Comment by rkehoe** [comment]
@johnridge I believe the PRs relevant for this issue have been merged.  I am closing this issue.
